src/app/lib/utilities/passport/index.js

```diff
@@ -20,15 +19,15 @@ export const isValidPassportHome = (
   service,
   passportHomesOverride = [],
 ) => {
-  const passportHomesOverrideArray = passportHomesOverride || [];
-  const passportHomeLower = (passportHome || '').toLowerCase();
+  const isMissingRequiredArgs = !passportHome && !passportHomesOverride.length;
+  if (isMissingRequiredArgs) return true;
 
-  if (!passportHome && !passportHomesOverrideArray.length) return true;
+  const matchesPassportHome = (home = '') =>
+    (home || '').toLowerCase() === passportHome.toLowerCase();
 
-  return (
-    passportHomeLower === service ||
-    passportHomesOverrideArray.some(
-      home => (home || '').toLowerCase() === passportHomeLower,
-    )
-  );
+  if (matchesPassportHome(service)) return true;
+
+  const passportHomesOverrideArray = passportHomesOverride || [];
```

Comment:
why do we need `passportHomesOverrideArray`?

Can you not just do `return passportHomesOverride.some(matchesPassportHome);`

Reply:
`passportHomesOverride` could be explicitly set to null, in which case it would throw an error.

Reply:
Ah ok. Would be better if it was left undefined rather than explicitly setting to null. Anyway, if it is ever set to null then this function would error on line 22 with `!passportHomesOverride.length`

Reply:
We've taken out the `!passportHomesOverride.length` on line 22, because the condition should only check that `passportHome` is null.

See [this PR](https://github.com/bbc/simorgh/pull/3967/files) from yesterday.